### PR TITLE
Feature/adgprs plugin dev from ov

### PR DESCRIPTION
**Pull request by Bellout**
Primarily introducing changes needed to compile WIC  as independent plugin for ADGPRS. Needs proper testing by OV (within Visual Basic platform), but eventual fixes can be introduced in subsequent PRs.

### CODE REVIEW

File: main.hpp [hunk 1]
```diff
@@ -47,82 +60,149 @@ void printCsv(vector<IntersectedCell> &well_blocks) {
...
-    assert(vm.count("heel"));
-    assert(vm.count("toe"));
-    assert(vm.count("radius"));
-    if (vm.count("compdat"))
-        assert(vm.count("well-name"));
-    assert(vm["heel"].as<vector<double>>().size() == 3);
-    assert(vm["toe"].as<vector<double>>().size() == 3);
-    assert(boost::filesystem::exists(vm["grid"].as<string>()));
-    assert(vm["radius"].as<double>() > 0);
+    // Check input parameters are well-defined
```
Comment (einar90) [inline main.hpp]: I'm really not a fan of this way of checking the input. This is exactly what `assert` is for. If you want to print a custom message, this should work: `assert(vm.count("grid") && "grid parameter is missing..." + usage_msg);`. The output will look like the following:
```
wicalc: /home/einar/Documents/GitHub/PCG/FieldOpt/FieldOpt/FieldOpt-WellIndexCalculator/main.hpp:144: boost::program_options::variables_map createVariablesMap(int, const char**): Assertion `vm.count("grid") && "Grid is missing"' failed.
[1]    5100 abort (core dumped)  ./wicalc
```
The message may be a bit messy, but it has the advantage of saying exactly where in the code the error occured. 

Alternatively, a way I actually like more than both is to create a custom assert macro that lets us specify the output format, while still being as easy to use as the standard `assert` macro. An example of how to implement one is given in [this answer](http://stackoverflow.com/a/37264642).

Also, indentation is a bit off. Use auto indent.

File: wellindexcalculator.cpp [hunk 2]
```diff
@@ -27,32 +27,39 @@
...
 
-        std::vector<IntersectedCell> WellIndexCalculator::ComputeWellBlocks(Vector3d heel, Vector3d toe, double wellbore_radius) {
-            heel_ = heel;
-            toe_ = toe;
+        vector<IntersectedCell>
+        WellIndexCalculator::ComputeWellBlocks(Vector3d heel,
+                                                Vector3d toe,
+                                                double wellbore_radius) {
+                       heel_ = heel;
+                        toe_ = toe;
```
Comment (einar90) [inline wellindexcalculator.cpp]: Indentation. Use auto indent.

File: wellindexcalculator.cpp [hunk 3]
```diff
@@ -27,32 +27,39 @@
...
             wellbore_radius_ = wellbore_radius;
+            vector<IntersectedCell> intersected_cells = cells_intersected();
 
-            std::vector<IntersectedCell> intersected_cells = cells_intersected();
-	    
             for (int i = 0; i < intersected_cells.size(); ++i) {
-                intersected_cells[i].set_well_index(compute_well_index(intersected_cells[i]));
+                intersected_cells[i].set_well_index(
```
Comment (einar90) [inline wellindexcalculator.cpp]: IMO, splitting up a simple and relatively short call like this makes it less readable.

File: wellindexcalculator.cpp [hunk 4]
```diff
@@ -62,20 +69,31 @@ namespace Reservoir {
...
             // Make sure we follow line in the correct direction. (i.e. dot product positive)
-            Vector3d exit_point = find_exit_point(intersected_cells[0], heel_, toe_, heel_);
+            Vector3d exit_point;
+            exit_point = find_exit_point(intersected_cells[0],
+                                         heel_,
+                                         toe_,
+                                         heel_);
+
             if ((toe_ - heel_).dot(exit_point - heel_) <= 0.0) {
-                exit_point = find_exit_point(intersected_cells[0], heel_, toe_, exit_point);
```
Comment (einar90) [inline wellindexcalculator.cpp]: Same as above.

File: wellindexcalculator.cpp [hunk 5]
```diff
@@ -84,25 +102,32 @@ namespace Reservoir {
                 }
 
                 // Find the exit point of the cell and set it in the list
-                exit_point = find_exit_point(intersected_cells.back(), exit_point, toe_, exit_point);
+                exit_point = find_exit_point(intersected_cells.back(),
```
Comment (einar90) [inline wellindexcalculator.cpp]: and here.

File: dllmain.hpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
+// dllmain.h : include file for standard system include files,
```
Comment (Bellout) [inline dllmain.hpp]: Skipping this for now since this is OVs file.

File: dllmain.cpp [hunk 7]
```diff
@@ -0,0 +1,135 @@
+
```
Comment (Bellout) [inline dllmain.cpp]: Skipping this for now since this is OVs file.

File: intersected_cell.cpp [hunk 8]
```diff
@@ -1,3 +1,23 @@
+/******************************************************************************
+   Copyright (C) 2015-2016 Einar J.M. Baumann <einar.baumann@gmail.com>
+???? Copyright (C) 2015-2016 Hilmar M. Magnusson <hilmarmag@gmail.com>
```
Comment (Bellout) [inline intersected_cell.cpp]: Not sure if Hilmar should be added here, should he?

File: main.cpp [hunk 9]
```diff
@@ -22,38 +22,43 @@
...
 int main(int argc, const char *argv[]) {
     // Initialize some variables from the runtime arguments
-    auto vm = createVariablesMap(argc, argv);
-	
-    auto heel = Eigen::Vector3d(vm["heel"].as<vector<double>>().data());
-    auto toe = Eigen::Vector3d(vm["toe"].as<vector<double>>().data());
-    string gridpth = vm["grid"].as<string>();
+    Eigen::setNbThreads(1); // OV
+
+                   auto vm = createVariablesMap(argc, argv);
```
Comment (Bellout) [inline main.cpp]: Fixed

File: main.cpp [hunk 10]
```diff
@@ -22,38 +22,43 @@
...
+ * @brief This file contains the main function for the stand-alone
+ * well index calculator executable.
  */
 
 #include "main.hpp"
 #include "wellindexcalculator.h"
 #include <Reservoir/grid/eclgrid.h>
 
 using namespace std;
+using Eigen::Vector3d;
```
Comment (Bellout) [inline main.cpp]: Removed it altogether, it doesn't seem to be needed at all in this file, maybe it was a hold over for something I was trying...

File: main.hpp (hunk 1 above)
Comment (Bellout) [inline main.hpp]: Fixed indentation. I suggest we leave further changes, e.g., building an assert macro, for later versions, or when upgrading the FieldOpt executable, if necessary. This is not critical right now.

File: wellindexcalculator.cpp (hunk 2 above)
Comment (Bellout) [inline wellindexcalculator.cpp]: Fixed w/ auto indent.

File: wellindexcalculator.cpp (hunk 3 above)
Comment (Bellout) [inline wellindexcalculator.cpp]: Ok, reverted to one line format given the new indenting makes the line just wide enough.

File: wellindexcalculator.cpp (hunk 4 above)
Comment (Bellout) [inline wellindexcalculator.cpp]: Ok, fixed.

File: wellindexcalculator.cpp (hunk 5 above)
Comment (Bellout) [inline wellindexcalculator.cpp]: Ok, fixed.

File: intersected_cell.cpp (hunk 8 above)
Comment (einar90) [inline intersected_cell.cpp]: Well... I created and wrote the file, so I guess not, but it's not that important.

**Comment by Bellout** [comment]
Can you delete this branch?

**Comment by Bellout** [comment]
Wait 2secs

**Pull request merged**